doc/TAEF.md

```diff
@@ -1,9 +1,28 @@
-### TAEF ###
+### TAEF Overview ###
+
 TAEF, the Test Authoring and Execution Framework, is used extensively within the Windows organization to test the operating system code in a unified manner for system, driver, and application code. As the console is a Windows OS Component, we strive to continue using the same system such that tests can be ran in a unified manner both externally to Microsoft as well as inside the official OS Build/Test system.
 
-The [official documentation](https://msdn.microsoft.com/en-us/library/windows/hardware/hh439725\(v=vs.85\).aspx) for TAEF describes the basic architecture, usage, and functionality of the test system. It is similar to Visual Studio test, but a bit more comprehensive and flexible.
+The [official documentation](https://docs.microsoft.com/en-us/windows-hardware/drivers/taef/) for TAEF describes the basic architecture, usage, and functionality of the test system. It is similar to Visual Studio test, but a bit more comprehensive and flexible.
+
+### Writing Tests
+
+You may want to read the section [Authoring Tests in C++](https://docs.microsoft.com/en-us/windows-hardware/drivers/taef/authoring-tests-in-c--) before getting your hands dirty. Note that the quoted header name in `#include "WexTestClass.h"` might be a bit confusing. You are not required to copy TAEF headers into the project folder.
+
+Use the [TAEF Verify Macros for C++](https://docs.microsoft.com/en-us/windows-hardware/drivers/taef/verify) in your test code to perform verifications.
+
+### Running Tests
+
+If you have Visual Studio and related C++ components installed, you should have the TAEF test runner `te.exe` available locally.
+
+In a "normal" command line enviroment, `te.exe` may not be directly available. Try the following command to set up the development enviroment first:
+
+```shell
+.\tools\razzle.cmd
+```

```

Comment:
Do note that if you use PowerShell and `Import-Module .\tools\OpenConsole.psm1`, you will then be able to do `Invoke-OpenConsoleTests` (optionally with a few switches you can read inside that PowerShell script) and it will get things going for you as an alternative to the CMD way.